### PR TITLE
feat: bump minimum go version to 1.18

### PR DESCRIPTION
**Pull request by padamstx**
This commit modifies the project so that our mininum supported version of Go is now 1.18.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -30,17 +31,28 @@ If you want to contribute to the repository, here's a quick guide:
...
   that the automated build uses.  
   
-  5. Make your code changes as needed.  Be sure to add new tests for any new or modified functionality.  
+  5. Install the `gosec` tool:
+     ```sh
+     curl -sfL https://raw.githubusercontent.com/securego/gosec/master/install.sh | sh -s -- -b $(go env GOPATH)/bin
+     ```  
+ 
+  6. Make sure that you have a clean "baseline" to work from by building/testing the project before
+     you start to make chagnes:
```
Comment (pyrooka) [inline CONTRIBUTING.md:48]:
```suggestion
     you start to make changes:
```
👀

**Review by pyrooka — APPROVED** [review]
Looks good!

**Review by dpopp07 — APPROVED** [review]
LGTM!

File: CONTRIBUTING.md (hunk 1 above)
Comment (padamstx) [reply CONTRIBUTING.md:48]: Good catch!

**Comment by padamstx** [comment]
I'm re-requesting reviews because I just pushed an amended commit that removes the v5 directory. I've been thinking about that change for a while and now that we're moving up to a new go version, it seemed like the right time to do it.

**Review by pyrooka — APPROVED** [review]
Looks good!

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍

**Pull request merged**